doc/builders/images/binarycache.section.md

```diff
@@ -0,0 +1,32 @@
+# pkgs.mkBinaryCache {#sec-pkgs-binary-cache}
+
+`pkgs.mkBinaryCache` is a function for creating Nix flat-file binary caches. Such a cache exists as a directory on disk, and can be used as a Nix substituter by passing `--substituter file:///path/to/cache` to Nix commands.
+
+Nix packages are most commonly shared between machines using [HTTP, SSH, or S3](https://nixos.org/manual/nix/stable/package-management/sharing-packages.html), but a flat-file binary cache can still be useful in some situations. For example, you can copy it directly to another machine, or make it available on a network file system. It can also be a convenient way to make some Nix packages available inside a container via bind-mounting, when you want to efficiently run Nix builds inside a container without doing a writable mount of the host's `/nix` directory.
```

Comment:
After reading this, I'd almost be convinced that this is the way to do binary caches on filesystems. It is not, and I would not recommend to use it except in very specific cases. It should at least say that the idiomatic way to work with an on-filesystem cache is through `nix-copy-closure` or `nix copy`, because it is more efficient and flexible that way.

> when you want to efficiently run Nix builds inside a container without doing a writable mount of the host's `/nix` directory.

`dockerTools` can already do this for you. It will put the packages in the local store rather than a separate directory, so no copying is necessary. This is arguably more efficient. What constraints might make the `dockerTools` solution infeasible for someone?

Reply:
> What constraints might make the dockerTools solution infeasible for someone?

That would require you to know in advance what you want baked into the store. The use-case of a bind-mounted binary cache is that you can mount different binary caches onto a fixed Docker image. Why you might want to do this is a bit of a rabbit hole, I think I'll just shorten this section to not mention it.

Reply:
Okay, I'm done with another pass on the docs. Please take a look!